api/server/router/network/network_routes.go

```diff
@@ -139,20 +132,22 @@ func (n *networkRouter) getNetwork(ctx context.Context, w http.ResponseWriter, r
 	listByFullName := map[string]types.NetworkResource{}
 	listByPartialID := map[string]types.NetworkResource{}
 
-	nw := n.backend.GetNetworks()
+	// TODO(@cpuguy83): All this logic for figuring out which network to return does not belong here
+	// Instead there should be a backend function to just get one network.
+	nw := n.backend.GetNetworks(types.NetworkListConfig{Detailed: true, Verbose: verbose})
 	for _, network := range nw {
-		if network.ID() == term && isMatchingScope(network.Info().Scope(), scope) {
-			return httputils.WriteJSON(w, http.StatusOK, *n.buildDetailedNetworkResources(network, verbose))
+		if network.ID == term && isMatchingScope(network.Scope, scope) {
```

Comment:
Can you put the `isMatchingScope(network.Scope, scope)` for all of these at the start? i.e.;

```go
for _, network := range nw {
    if !isMatchingScope(network.Scope, scope) {
        continue
    }
    if network.ID == term {
        return httputils.WriteJSON(w, http.StatusOK, network)
    }
    ....

Reply:
Even better, handling this in the filter logic.